applications/serial_lte_modem/src/slm_at_udp_proxy.c

```diff
@@ -210,6 +230,34 @@ static int do_udp_client_connect(const char *url, uint16_t port)
 		goto cli_exit;
 	}
 
+	/* Check conn_id status */
+	if (proxy.conn_id != 0) {
+		int status = -1;
+		int status_len = sizeof(status);
+
+		ret = nrf_getsockopt(proxy.sock, NRF_SOL_SECURE, NRF_SO_SEC_DTLS_CID_STATUS,
+				     &status, &status_len);
+		if (ret) {
+			LOG_WRN("Unable to get DTLS connection ID status: %d", ret);
+		} else {
+			/**
+			 * NRF_SO_SEC_DTLS_CID_STATUS_DISABLED 0
+			 * NRF_SO_SEC_DTLS_CID_STATUS_DOWNLINK 1
+			 * NRF_SO_SEC_DTLS_CID_STATUS_UPLINK 2
+			 * NRF_SO_SEC_DTLS_CID_STATUS_BIDIRECTIONAL 3
+			 */
+			LOG_INF("DTLS connection ID status: %d", status);
+			if (status != NRF_SO_SEC_DTLS_CID_STATUS_DISABLED) {
+				ret = nrf_setsockopt(proxy.sock, NRF_SOL_SECURE,
+						     NRF_SO_SEC_DTLS_CONN_SAVE,
+						     NULL, 0);
+				if (ret) {
+					LOG_WRN("Save DTLS connection failed: %d", ret);
+				}
```

Comment:
Why do we blindly save the session if connection ID is enabled?
I would propose that we don't use that feature as it might run out of memory.

Reply:
I think there will be only ONE session saved on modem side, as only SAVE/LOAD defined and no DELETE, so there should be no incremental memeory usage. SAVE after confirmation of CID active would only update the session data as I assumed.